packages/core/src/extensions/Blocks/nodes/Block.ts

```diff
@@ -162,275 +109,408 @@ export const Block = Node.create<IBlock>({
-              tr.setNodeMarkup(nodePos, undefined, {
-                ...node.attrs,
-                listType: type,
-              });
-            }
-            return true;
+      // Changes the block at a given position to a given content type.
+      BNSetContentType:
+        (posInBlock, type, attributes) =>
+        ({ state }) => {
+          const blockInfo = getBlockInfoFromPos(state.doc, posInBlock);
+          if (blockInfo === undefined) {
+            return false;
+          }
+
+          const { startPos, endPos } = blockInfo;
+
+          state.tr.setBlockType(
+            startPos + 1,
+            endPos - 1,
+            state.schema.node(type).type,
+            attributes
+          );
+
+          return true;
+        },
+      // Changes the block at a given position to a given content type if it's empty, otherwise creates a new block of
+      // that type below it.
+      BNCreateBlockOrSetContentType:
+        (posInBlock, type, attributes) =>
+        ({ state, chain }) => {
+          const blockInfo = getBlockInfoFromPos(state.doc, posInBlock);
+          if (blockInfo === undefined) {
+            return false;
+          }
+
+          const { node, startPos, endPos } = blockInfo;
+
+          if (node.textContent.length === 0) {
+            const oldBlockContentPos = startPos + 1;
+
+            return chain()
+              .BNSetContentType(posInBlock, type, attributes)
+              .setTextSelection(oldBlockContentPos)
+              .run();
+          } else {
+            const newBlockInsertionPos = endPos + 1;
+            const newBlockContentPos = newBlockInsertionPos + 1;
+
+            return chain()
+              .BNCreateBlock(newBlockInsertionPos)
+              .BNSetContentType(newBlockContentPos, type, attributes)
+              .setTextSelection(newBlockContentPos)
+              .run();
           }
-          return false;
         },
-      joinBackward:
-        () =>
-        ({ view, dispatch, state }) =>
-          joinBackward(state, dispatch, view), // Override default joinBackward with edited command
     };
   },
+
   addProseMirrorPlugins() {
-    return [PreviousBlockTypePlugin(), OrderedListPlugin()];
+    return [PreviousBlockTypePlugin()];
   },
+
   addKeyboardShortcuts() {
     // handleBackspace is partially adapted from https://github.com/ueberdosis/tiptap/blob/ed56337470efb4fd277128ab7ef792b37cfae992/packages/core/src/extensions/keymap.ts
     const handleBackspace = () =>
       this.editor.commands.first(({ commands }) => [
-        // Maybe the user wants to undo an auto formatting input rule (e.g.: - or #, and then hit backspace) (source: tiptap)
+        // Deletes the selection if it's not empty.
+        () => commands.deleteSelection(),
+        // Undoes an input rule if one was triggered in the last editor state change.
         () => commands.undoInputRule(),
-        // maybe convert first text block node to default node (source: tiptap)
```

Comment:
this functionality was removed, it seems?

Reply:
I don't really understand what this does. As far as I understand, calls `clearNodes` if the selection is empty and at the start of the doc, and if the block it's in is empty. But TipTap docs say that `clearNodes` changes all node types to "what’s in the content attribute of your Document", which in our case is a blockGroup, so I'm not sure what effect it's supposed to have on the editor.

Reply:
I think this:
![firstbackspace](https://user-images.githubusercontent.com/368857/204845338-5712d3db-072f-4a09-af04-57afc1fded87.gif)